src/solver/qualrout.c

```diff
@@ -128,11 +130,15 @@ void qualrout_execute(double tStep)
         else findNodeQual(j);
 
         // --- apply treatment to new quality values
-        if ( Node[j].treatment ) treatmnt_treat(j, qIn, vAvg, tStep);
+        if ( Node[j].treatment || ExtPollutFlag == 1) treatmnt_treat(j, qIn, vAvg, tStep);
     }
 
     // --- find new water quality in each link
     for ( i = 0; i < Nobjects[LINK]; i++ ) findLinkQual(i, tStep);
+
+    // --- default to swmm treatment
+    if (ExtPollutFlag == 1){ExtPollutFlag = 0;}
```

Comment:
Can you explain the purpose of this? Where is ExtPollutFlag defined? What is it turned off if it is was turned on previously.

Reply:
For sure, ExtPollutFlag is created when a pollutant object is created for the node or link. It controls what pollutant to set in the node. If 0, which is set by default, it uses the pollutant routing in swmm and when 1 it sets newqual to the value in extqual.

As the end of each step, we reset flag to 0 so that if user does not provide a pollutant then it go back to swmm routing than breaking down.